modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/js/ckeditor5/plugins/ItemSelector.tsx

```diff
@@ -0,0 +1,58 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {ButtonView, Config, Plugin, icons} from 'ckeditor5';
+import {openSelectionModal} from 'frontend-js-components-web';
+
+import {ClassicEditorConfig} from '../utils/types';
+
+class ItemSelector extends Plugin {
+	init() {
+		const editor = this.editor;
+
+		const config: Config<ClassicEditorConfig> = editor.config;
+
+		editor.ui.componentFactory.add('imageSelector', () => {
+			const buttonView = new ButtonView();
+
+			buttonView.set({
+				icon: icons.image,
+				label: Liferay.Language.get('image'),
+				tooltip: true,
+			});
+
+			buttonView.on('execute', () => {
+				openSelectionModal({
+					onSelect: ({value}: {value: string}) => {
+						let url;
+
+						try {
+							url = JSON.parse(value).url;
+						}
+						catch (error) {
+							url = value;
+						}
```

Comment:
Handling quirks of `openSelectionModal`. I ran into both of these cases in testing, one in Web Content and one in the sample module.